index.bs

```diff
@@ -215,10 +373,28 @@ Every [=Document=] has an associated [=ordered set=] of <dfn>previously reported
         1. Set |newEntry|'s {{PerformanceEntry/startTime}} attribute to |paintTimestamp|.
         1. Set |newEntry|'s {{PerformanceEntry/duration}} attribute to 0.
     1. <a href="https://w3c.github.io/performance-timeline/#dfn-queue-a-performanceentry">Add the PerformanceEntry</a> |newEntry| object.
-    1. [=list/Append=] |paintType| to |document|'s associated [=previously reported paints=].
+    1. [=list/Append=] |paintType| to |document|'s [=set of previously reported paints=].
+</div>
+
+
+Common algorithms {#sec-common-algorithms}
+------------------------------------------
+
+<h4 dfn export>Exposed for paint timing</h4>
+
+<div algorithm="Exposed for paint timing">
+
+    To determine whether an [=/Element=] |element| is [=exposed for paint timing=], given a [=Document=] or null |document|, perform the following steps:
+
+    1. If |element| is not [=connected=], return false.
+    1. If |document| is null, let |document| be |element|'s [=relevant settings object=]'s [=relevant global object=]'s [=associated document=].
+    1. If |document| is not [=fully active=], return false.
+    1. If |element|'s [=tree/root=] is not equal to |document|, return false.
+    1. Return true.
 </div>
 
 
+
 <!-- ============================================================ -->
 <h2 id=acknowledgements>Acknowledgements</h2>
```

Comment:
?

Reply:
Still not resolved

Reply:
Unsure how to resolve "`?`". Please advise.

Reply:
Oh sorry I was seeing something that wasn't there because of the red thingy.